R/ddi.R

```diff
@@ -142,6 +162,36 @@ DDI <- R6::R6Class(
       self$data <- private$read_json(self$source)
     }
   ),
-  private = list(),
+  private = list(
+    # Validate that all inputs are of class 'Compound'
+    # @param compounds A list of compound objects to validate.
+    validate_compounds = function(compounds) {
+      is_compound <- vapply(
+        compounds, \(comp) inherits(comp, "Compound"), logical(1)
+      )
+
+      if (!all(is_compound)) {
+        invalid_indices <- which(!is_compound)
+        invalid_classes <- vapply(compounds[invalid_indices], class, character(1))
+
+        invalid_details <- paste0("[", invalid_indices, "] ", invalid_classes)
+        cli::cli_abort(c(
+          "All compounds must be of class 'Compound'. Invalid entries found at position(s):",
+          invalid_details
+        ))
+      }
+
+      # create and print status message
```

Comment:
I think this is out of scope of the method. It can be moved (and simplified) to the combined method, outputing something like:
```
Combining Snapshots:
- Victim Compound: XXX
- Perpertrator(s) Compound(s):
	- YYY
	- ZZZ
```